lms/djangoapps/grades/api.py

```diff
@@ -108,8 +108,10 @@ def undo_override_subsection_grade(user_id, course_key_or_id, usage_key_or_id, f
         return
 
     if override is not None and (
-            not feature or feature == override.system):
+            not feature or not override.system or feature == override.system):
```

Comment:
This is difficult to parse.  Can we DeMorgan this into:
```python
if not override:
    return
if feature and override.system and (feature != override.system):
    return
else:
    override.delete(feature=feature)
```

Reply:
Merged because my tests just passed after addressing Nimisha's concern & the time of day.

Reply:
👍 I consider my suggestion a nice-to-have.  If you ever want to bump your edx-platform PR count metric, this might be a quick win :)